pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactorMXBeanImpl.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.compaction;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+public class CompactorMXBeanImpl implements CompactorMXBean {
+
+    private final ConcurrentHashMap<String, CompactRecord> compactRecordOps = new ConcurrentHashMap<>();
+
+    public void addCompactRemovedEvent(String topic) {
+        compactRecordOps.computeIfAbsent(topic, k -> new CompactRecord()).addCompactRemovedEvent();
+    }
+
+    public void addCompactStartOp(String topic) {
+        compactRecordOps.computeIfAbsent(topic, k -> new CompactRecord()).reset();
+    }
+
+    public void addCompactEndOp(String topic, boolean succeed) {
+        CompactRecord compactRecord = compactRecordOps.computeIfAbsent(topic, k -> new CompactRecord());
+        compactRecord.lastCompactDurationTimeInMills = TimeUnit.NANOSECONDS.toMillis(
+                System.nanoTime() - compactRecord.lastCompactStartTimeOp);
```

Comment:
Nit: why compute the difference in nano time if we're only going to capture millis in the metric? Seems like we could use `System.currentTimeMillis()` instead, and then we wouldn't need to convert time units here.

Reply:
Ok, thanks for kindly review.